extension/parquet/parquet_extension.cpp

```diff
@@ -457,11 +485,19 @@ class ParquetScanFunction {
 
 	static unique_ptr<FunctionData> ParquetScanBind(ClientContext &context, TableFunctionBindInput &input,
 	                                                vector<LogicalType> &return_types, vector<string> &names) {
-		auto files = MultiFileReader::GetFileList(context, input.inputs[0], "Parquet");
+		unique_ptr<MultiFileReader> multi_file_reader;
+		if (input.table_function.get_multi_file_reader) {
+			// Use the MultiFileReader from the Table Function
+			multi_file_reader = input.table_function.get_multi_file_reader();
```

Comment:
More of a question:
Do we perhaps want to already future proof this callback with a GetMultiFileReaderInput struct?
I can imagine this might want some context in the future

Reply:
Thanks for the comment, Mark said the same, further discussion is here: https://github.com/duckdb/duckdb/pull/11806#discussion_r1584653154